packages/changed-elements-react/src/widgets/comparisonJobWidget/components/VersionCompareSelectModal.tsx

```diff
@@ -0,0 +1,401 @@
+      targetVersion: args.targetVersion,
+      comparisonJobClient: args.comparisonJobClient,
+      iModelConnection: args.iModelConnection,
+      currentVersion: currentVersion,
+      getDialogOpen: args.getDialogOpen,
+    }).catch((e) => {
+      Logger.logError(VersionCompare.logCategory, "Could not start version comparison: " + e);
+    });
+  }
+};
+
+type RunStartComparisonV2Args = {
+  targetVersion: NamedVersion;
+  comparisonJobClient: IComparisonJobClient;
+  iModelConnection: IModelConnection;
+  currentVersion: NamedVersion;
+  getDialogOpen: () => boolean;
+};
+
+const runStartComparisonV2 = async (args: RunStartComparisonV2Args) => {
+  const { startedComparison } = await createOrRunManagerStartComparisonV2(args);
+  if (startedComparison) {
+    return;
+  }
+  toastComparisonJobProcessing(args.currentVersion, args.targetVersion);
+  void pollForComparisonJobTillComplete(args);
+};
+
+type PostOrRunComparisonJobResult = {
+  startedComparison: boolean;
+};
+
+const createOrRunManagerStartComparisonV2 = async (args: RunStartComparisonV2Args): Promise<PostOrRunComparisonJobResult> => {
+  const { comparisonJob } = await postOrGetComparisonJob({
+    changedElementsClient: args.comparisonJobClient,
+    iTwinId: args.iModelConnection?.iTwinId as string,
+    iModelId: args.iModelConnection?.iModelId as string,
+    startChangesetId: args.targetVersion.changesetId as string,
+    endChangesetId: args.currentVersion.changesetId as string,
+  });
+  if (comparisonJob.status === "Completed") {
+    void runManagerStartComparisonV2({
+      comparisonJob: { comparisonJob: comparisonJob },
+      comparisonJobClient: args.comparisonJobClient,
+      iModelConnection: args.iModelConnection,
+      targetVersion: args.targetVersion,
+      currentVersion: args.currentVersion,
+    });
+    return { startedComparison: true };
+  }
+  return { startedComparison: false };
+};
+
+const pollForComparisonJobTillComplete = async (args: RunStartComparisonV2Args) => {
+  const getComparisonJob = async () => {
+    const job = (await args.comparisonJobClient.getComparisonJob({
+      iTwinId: args.iModelConnection?.iTwinId as string,
+      iModelId: args.iModelConnection?.iModelId as string,
+      jobId: `${args.targetVersion.changesetId}-${args.currentVersion.changesetId}`,
+      headers: {
+        "Content-Type": "application/json",
+      },
+    })).comparisonJob;
+    return job;
+  };
+
+  let comparisonJob = await getComparisonJob();
+  let isConnectionClosed = false;
+  args.iModelConnection.onClose.addListener(() => { isConnectionClosed = true; });
+  const loopDelayInMilliseconds = 5000;
+
+  while (comparisonJob.status !== "Error" && !isConnectionClosed) {
+    await new Promise((resolve) => setTimeout(resolve, loopDelayInMilliseconds)); // run loop every 5 seconds
+    if (VersionCompare.manager?.isComparing) {
+      return;
+    }
+    comparisonJob = await getComparisonJob();
+    if (comparisonJob.status === "Completed") {
```

Comment:
iModelConnection may have been closed at this point but not at the start of the loop.